srt-protocol/src/protocol/time/base.rs

```diff
@@ -0,0 +1,135 @@
+use std::time::Instant;
+
+use crate::packet::{TimeSpan, TimeStamp};
+
+#[derive(Copy, Clone, Debug)]
+pub struct TimeBase {
+    // this field is only here for diagnostics and debugging
+    // it is similar to start_time in other contexts, but is adjusted for drift
+    origin_time: Instant,
+
+    // the two "reference" fields are two equivalent time points from different
+    // time scales they are reference points used for mapping between the Instance
+    // and a sender TimeStamp time scales
+    reference_time: Instant,
+    reference_ts: TimeStamp,
+}
+
+impl TimeBase {
+    pub fn new(start_time: Instant) -> Self {
+        Self {
+            origin_time: start_time,
+            reference_time: start_time,
+            reference_ts: TimeStamp::MIN,
+        }
+    }
+
+    #[allow(clippy::trivially_copy_pass_by_ref)]
```

Comment:
Why suppress this? Can't we just pass by `self`?

Reply:
I think this was copied from elsewhere, I am unsure why this was here, I imagine just to appease clippy at some point.